backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt

```diff
@@ -365,17 +365,17 @@ internal val allLoweringsPhase = NamedCompilerPhase(
                             localFunctionsPhase,
                             foldConstantLoweringPhase,
                             tailrecPhase,
+                            singleAbstractMethodPhase,
```

Comment:
What is this reording required for?

Reply:
For handling SAM conversions to function references (which is done in `FunctionReferenceLowering`).

Reply:
This explains placing `singleAbstractMethodPhase` after `functionReferencePhase`.
I'm asking about moving bunch of lowerings in e3386e26b15a23f55194683cde4ac6366a4237af.

Reply:
Looks like other movements not needed (all the relevant tests seem to be passing), probably this was an artefact of some experiments.

Reply:
Reordered. The final order reflects this chain of dependencies: BuiltInOperatorLowering -> SAMLowering -> FunRefLowering -> LocalDeclLowering.

Reply:
Should this affect `testProcessorPhase`?

Reply:
Ah, I see.